src/boot/bootctl.c

```diff
@@ -2555,6 +2556,334 @@ static int verb_reboot_to_firmware(int argc, char *argv[], void *userdata) {
+                        if (!entry->options[i])
+                                break;
+
+                        s = strpush(s, "options    ", entry->options[i], "\n");
+                        if (!s) {
+                                free(s);
+                                return NULL;
+                        }
+                }
+        }
+
+        if (entry->initrd) {
+                for (size_t i = 0; i < SIZE_MAX; i++)
+                {
+                        if (!entry->initrd[i])
+                                break;
+
+                        s = strpush(s, "initrd    ", entry->initrd[i], "\n");
+                        if (!s) {
+                                free(s);
+                                return NULL;
+                        }
+                }
+        }
+
+        return s;
+}
+
+/* sort_key, machine_id, title, version and options must be freed. See boot_entry_auto_done */
+static int boot_entry_auto(BootEntry* ret_entry, char* kernel, char** initrds) {
+        int r;
+
+        char* os_release_pretty_name = NULL;
+        char* os_release_name = NULL;
+        char* os_release_id = NULL;
+        char* os_release_version = NULL;
+        char* machine_id = NULL;
+        char* cmdline = NULL;
+        char** options = NULL;
+
+        /* Read /etc/os-release */
+        r = parse_os_release("/",
+                "PRETTY_NAME", &os_release_pretty_name,
+                "NAME", &os_release_name,
+                "ID", &os_release_id,
+                "VERSION", &os_release_version
+        );
+        if (r < 0)
+                goto error;
+
+        /* Read /etc/machine-id */
+        r = load_etc_machine_id();
+        if (r < 0)
+                goto error;
+
+        machine_id = new(char, 33);
+        if (!machine_id) {
+                r = -ENOMEM;
+                goto error;
+        }
+        sd_id128_to_string(arg_machine_id, machine_id);
+
+        /* Read /proc/cmdline */
+        size_t cmdline_size = 0;
+        r = read_full_file("/proc/cmdline", &cmdline, &cmdline_size);
+        if (r < 0)
+                goto error;
+
+        /* Remove the final new line if any */
+        char* c = strrchr(cmdline, '\n');
+        if (c)
+                *c = '\0';
+
+
+        if (!os_release_id) {
+                os_release_id = strdup("linux");
+                if (!os_release_id) {
+                        r = -ENOMEM;
+                        goto error;
+                }
+        }
+
+        if (!os_release_name) {
+                os_release_name = strdup("Unknown Linux");
+                if (!os_release_name) {
+                        r = -ENOMEM;
+                        goto error;
+                }
+        }
+
+        if (!os_release_version) {
+                os_release_version = strdup("No Version Information");
+                if (!os_release_version) {
+                        r =-ENOMEM;
+                        goto error;
+                }
+        }
```

Comment:
It doesn't make sense to fill in fallback values like this. If something is missing, then it should be missing in the menu entry too. Every user of menu entries is supposed to implement suitable fallbacks.